serverless/serverless-kn-source.adoc

```diff
@@ -0,0 +1,45 @@
+[id="serverless-kn-source"]
+= Using Knative Eventing sources
+include::modules/common-attributes.adoc[]
+include::modules/serverless-document-attributes.adoc[]
+:context: serverless-kn-source
+
+You can use `kn` to list available event sources or event source types for use with Knative Eventing.
```

Comment:
Here, I would add a bit more explanations about source types and source. E.g.

One key element of Knative Eventing are _Sources_. A source is a resource object, which is the link between an event generating system like Kafka or the Kubernetes API Server and a so called _sink_, which receives those events. Examples for sinks are Knative services, Knative eventing channels or a Knative eventing broker. A source has a type which connected of the kind of events its imports. Out-of-the box, there are three sources available:

* `ApiServerSource` connects a sink to the Kubernetes API server for feeding such events into the sink
* `PingSource` creates periodically an event with a constant payload. It can be used as a timer.
* `SinkBinding` is a special kind of source we can connect core Kubernetes resources like `Deployment`, `Job` or `StatefulSet` with a sink.

Additional sources can be added by the cluster administrator. For example a `KafkaSource` pick ups messages from a Kafka cluster, transform this into CloudEvents and sends it to the configured sink.

In order to list all available types of sources, you can use `kn source list-type`. For each type you can create multiple instances of sources with different configurations (like two KafkaSources to connect to two different Kafka clusters). The relationship of a _source type_ to a _source_ is that of a Java class to a Java object, or a container image to a container. 

For listing all source instances use `kn source list`

Reply:
Feel free to adapt this text, shorten it or move it to somewhere (or ignore it ;)